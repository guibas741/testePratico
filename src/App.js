import React from 'react';
import Routes from './routes';
import {Provider} from 'react-redux';
import ReduxThunk from 'redux-thunk';
import {createStore, applyMiddleware} from 'redux';
import appReducer from './ducks/store';

const App = () => {
  const store = createStore(appReducer, {}, applyMiddleware(ReduxThunk));
  return (
    <Provider store={store}>
      <Routes />
    </Provider>
  );
};

export default App;
