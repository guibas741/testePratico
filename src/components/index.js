export * from './common/simpleButton';
export * from './common/title';
export * from './common/input';
