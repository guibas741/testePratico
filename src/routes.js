import {createAppContainer} from 'react-navigation';
import {createStackNavigator} from 'react-navigation-stack';
import LoginScreen from './screens/loginScreen';
import HomeScreen from './screens/homeScreen';

const RoutesContainer = createStackNavigator({
  Login: {
    screen: LoginScreen,
    navigationOptions: {
      header: null,
    },
  },
  Home: {
    screen: HomeScreen,
    navigationOptions: {
      header: null,
    },
  },
  /* { 
    initialRouteName: 'SplashScreen',
  } */
});

const Routes = createAppContainer(RoutesContainer);

export default Routes;
