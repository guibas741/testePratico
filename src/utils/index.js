export * from './scaling';
export * from './pickerHelper';
