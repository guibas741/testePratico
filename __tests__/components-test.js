import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import {SimpleButton, Spinner, Input, LabelWithText} from '../src/components';

it('<SimpleButton>', () => {
  const snap = renderer
    .create(<SimpleButton onPress={jest.fn()}>Clique</SimpleButton>)
    .toJSON();
  expect(snap).toMatchSnapshot();
});

it('<Spinner>', () => {
  const snap = renderer.create(<Spinner />).toJSON();
  expect(snap).toMatchSnapshot();
});

it('<Input>', () => {
  const snap = renderer
    .create(<Input onChangeText={jest.fn()} placeholder="test" value="" />)
    .toJSON();
  expect(snap).toMatchSnapshot();
});

it('<LabelWithText>', () => {
  const snap = renderer
    .create(<LabelWithText label="test" text="test" />)
    .toJSON();
  expect(snap).toMatchSnapshot();
});
