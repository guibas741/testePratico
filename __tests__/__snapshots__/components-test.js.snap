// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="test"
    placeholderTextColor="#d3d3d3"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#d3d3d3",
        "borderRadius": 10,
        "borderWidth": 2,
        "fontFamily": "sans-serif",
        "fontSize": 18.857142857142858,
        "marginBottom": 15.714285714285714,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`<LabelWithText> 1`] = `
<View
  style={
    Object {
      "margin": 7.857142857142857,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#d3d3d3",
        "fontFamily": "sans-serif",
        "fontSize": 18.857142857142858,
      }
    }
  >
    test
  </Text>
  <Text
    style={
      Object {
        "color": "#4D4E4F",
        "fontFamily": "sans-serif",
        "fontSize": 22,
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`<SimpleButton> 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItem": "center",
      "backgroundColor": "#0080FF",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 15.714285714285714,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontFamily": "sans-serif",
          "fontSize": 22,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Clique
  </Text>
</View>
`;

exports[`<Spinner> 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "margin": 3,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#0080FF"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
